### ISSUE TITLE
Rails 5.1 support

### ISSUE THREAD

**Issue by manuelmeurer** [chaadow/active_record-acts_as#8]
Rails 5.1 is not yet supported since one spec fails: https://travis-ci.org/krautcomputing/active_record-acts_as/jobs/258840517

I have tried to dig into ActiveRecord and find out what changed from 5.0 to 5.1 and how to fix it but failed. If anyone with more knowledge of the ActiveRecord internals could have a look, that'd be great! 🥇 

I created a [separate branch](https://github.com/krautcomputing/active_record-acts_as/tree/rails5.1) for Rails 5.1 compatibility.

**Comment by charliee22** [comment]
Hi,

if I change the migration file with the `t.actable` from 5.1 to 5.0

`ActiveRecord::Migration[5.0]` 

In my case it runs like before with rails 5.0
It is maybe a bad practise to mix 5.1 and 5.0 migrations. Perhaps it helps to localize the issue!

**Comment by charliee22** [comment]
I just realized that it drops this information when I insert new items in the parent table

`DEPRECATION WARNING: The behavior of `changed?` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method after `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_changes?` instead. (called from irb_binding at (irb):1)
DEPRECATION WARNING: The behavior of `changed_attributes` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method after `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_changes.transform_values(&:first)` instead. (called from irb_binding at (irb):1)`

**Comment by manuelmeurer** [comment]
Not sure what you mean, @charliee22...
What exactly did you change where?
Could you open a PR with your changes?

**Comment by jeremyjbarth** [comment]
I can confirm @charliee22's workaround.  If you change the migration version from `ActiveRecord::Migration[5.1]` to `ActiveRecord::Migration[5.0]`, the below works fine:
```
class AddActableToProducts < ActiveRecord::Migration[5.0]
  def change
    change_table :products do |t|
      t.actable
    end
  end
end
```

**Comment by manuelmeurer** [comment]
How does this fix the failing spec?
What's the error when you use `ActiveRecord::Migration[5.1]`?

**Comment by jeremyjbarth** [comment]
Sorry, I didn't run the specs.  I just tried the migration per @charliee22's suggestion.  I didn't even try it with `5.1`.

**Comment by etherbob** [comment]
active_record-acts_as/lib/active_record/acts_as/instance_methods.rb

Has a method `#changed?`

``` 
def changed?
  super || acting_as.changed? || (defined?(@_acting_as_changed) ? @_acting_as_changed : 
end
```

Adding the following seems to get the specs working again with Rails 5.1:
``` 
def saved_changes?
  super || acting_as.saved_changes? || (defined?(@_acting_as_changed) ? @_acting_as_changed : 
end
```

I'm not using this gem (yet) so I couldn't begin to tell you if the above fix is a good idea or not, but we might give it a kick locally as we're really wishing we had MTI support on a Rails 5.1 app.

**Comment by bidluo** [comment]
Is this still an issue? I've ran specs against 5.1 and 5.2 and both pass

**Comment by manuelmeurer** [comment]
Hmm, that's weird. I just restarted the tests and they still fail: https://travis-ci.org/krautcomputing/active_record-acts_as/jobs/258840517
Are you doing anything different, @bidluo?

**Comment by bidluo** [comment]
Pretty much did exactly the same as you did when the build started failing
Branch: https://github.com/bidluo/active_record-acts_as/pull/1
Build Status: https://travis-ci.org/bidluo/active_record-acts_as

Possibility of it being an environment issue? The failing builds are built with (I assume) Ubuntu Precise whereas my builds are using Trusty. Travis is decommissioning Precise anyway, so might be a good idea to switch anyway.
Have you tried locally building? Builds fine locally for me and I'm on slightly bleeding edge

**Comment by manuelmeurer** [comment]
Fixed in v3.0
